core/dbt/cli/main.py

```diff
@@ -139,11 +139,13 @@ def global_flags(func):
     @p.populate_cache
     @p.print
     @p.printer_width
+    @p.profile
     @p.quiet
     @p.record_timing_info
     @p.send_anonymous_usage_stats
     @p.single_threaded
     @p.static_parser
+    @p.target
```

Comment:
I think you'll also need to remove these (`@p.profile` and `@p.target`) where they're specified on specific subcommands down below. By adding them to `global_flags`, they are automatically supported both at the root level (`dbt --profile <subcommand>`) and after every subcommand (`dbt subcommand --profile`).

Reply:
@barton996 https://github.com/dbt-labs/dbt-core/pull/9081/commits/d404c9de4b8d40ec11105654eb0bae453f943f30 is intended to cover removing `@p.profile` and `@p.target` from all the subcommands, so hopefully you don't need to worry about this piece now.

Reply:
@dbeatty10 Thanks. I've also added a unit test to protect against this now

https://github.com/dbt-labs/dbt-core/blob/e925d2b0879c2116a8511a70870e5281ded735f7/tests/unit/test_cli.py#L56